Microsoft.Toolkit.HighPerformance/Enumerables/ReadOnlyRefEnumerable{T}.cs

```diff
@@ -121,6 +121,61 @@ internal ReadOnlyRefEnumerable(object? instance, IntPtr offset, int length, int
         }
 #endif
 
+        /// <summary>
+        /// Gets the total available length for the sequence.
+        /// </summary>
+        public int Length
+        {
+#if SPAN_RUNTIME_SUPPORT
+            get => this.span.Length;
+#else
+            get => this.length;
+#endif
+        }
+
+        /// <summary>
+        /// Gets the element at the specified zero-based index.
+        /// </summary>
+        /// <returns>A reference to the element at the specified index.</returns>
+        /// <exception cref="IndexOutOfRangeException">
+        /// Thrown when <paramref name="index"/> is invalid.
+        /// </exception>
+        public ref readonly T this[int index]
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get
+            {
+                if ((uint)index >= (uint)Length)
+                {
+                    ThrowHelper.ThrowIndexOutOfRangeException();
+                }
+
+#if SPAN_RUNTIME_SUPPORT
+                ref T r0 = ref MemoryMarshal.GetReference(this.span);
+#else
+                ref T r0 = ref RuntimeHelpers.GetObjectDataAtOffsetOrPointerReference<T>(this.instance, this.offset);
+#endif
+                nint offset = (nint)(uint)index * (nint)(uint)this.step;
+                ref T ri = ref Unsafe.Add(ref r0, offset);
+                return ref ri;
```

Comment:
Nit: new line before the return (for consistency with the rest of the codebase).
```suggestion

                return ref ri;
```